Makefile.dune

```diff
@@ -1,122 +1,109 @@
+	@echo ""
+	@echo " Provided opam/dune packages are:"
+	@echo ""
+	@echo "  - coq: base Coq package, toplevel compilers, tools, stdlib, no GTK"
+	@echo "  - coqide-server: XML protocol language server"
+	@echo "  - coqide: CoqIDE gtk application"
+	@echo ""
+	@echo " To build a package, you can use:"
+	@echo ""
+	@echo "  - 'dune build package.install' : build package in developer mode"
+	@echo "  - 'dune build -p package' : build package in release mode"
+	@echo ""
+	@echo " Packages _must_ be installed using release mode, use: 'dune install -p package'"
+	@echo " See Dune documentation for more information."
 
-# We need to bootstrap with a dummy coq.plugins.ltac so install targets do work.
-plugins/ltac/dune:
-	@echo "(library (name ltac_plugin) (public_name coq.plugins.ltac) (modules_without_implementation extraargs extratactics))" > plugins/ltac/dune
-
-voboot: plugins/ltac/dune
-	dune build --build-dir=$(BOOT_DIR) $(DUNEOPT) @vodeps
-	dune exec --build-dir=$(BOOT_DIR) -- ./tools/coq_dune.exe $(BOOT_CONTEXT)/.vfiles.d
+voboot:
+	@echo "This target is empty and not needed anymore"
 
-states: voboot
+states:
 	dune build --display=short $(DUNEOPT) dev/shim/coqtop-prelude
 
 NONDOC_INSTALL_TARGETS:=coq.install coqide-server.install coqide.install
 
-world: voboot
+world:
 	dune build $(DUNEOPT) $(NONDOC_INSTALL_TARGETS)
 
-coq: voboot
-	dune build $(DUNEOPT) coq.install
-
-coqide: voboot
-	dune build $(DUNEOPT) coqide.install
-
-coqide-server: voboot
-	dune build $(DUNEOPT) coqide-server.install
-
-watch: voboot
+watch:
 	dune build $(DUNEOPT) $(NONDOC_INSTALL_TARGETS) -w
 
-check: voboot
+check:
 	dune build $(DUNEOPT) @check
 
-COQTOP_FILES=ide/idetop.bc ide/coqide_main.bc checker/coqchk.bc
-PLUGIN_FILES=$(wildcard plugins/*/*.mlpack)
-PRINTER_FILES=dev/top_printers.cma
-QUICKBYTE_TARGETS=$(COQTOP_FILES) $(PLUGIN_FILES:.mlpack=.cma) $(PRINTER_FILES) topbin/coqtop_byte_bin.bc
-QUICKOPT_TARGETS=$(COQTOP_FILES:.bc=.exe) $(PLUGIN_FILES:.mlpack=.cmxs) $(PRINTER_FILES:.cma=.cmxa) topbin/coqtop_bin.exe
-
-quickbyte: voboot
-	dune build $(DUNEOPT) $(QUICKBYTE_TARGETS)
-
-quickopt: voboot
-	dune build $(DUNEOPT) $(QUICKOPT_TARGETS)
-
-quickide: states
-	dune build $(DUNEOPT) dev/shim/coqide-prelude
-
-test-suite: voboot
+test-suite:
 	dune runtest --no-buffer $(DUNEOPT)
 
-refman-html: voboot
+refman-html:
 	dune build @refman-html
 
-refman-pdf: voboot
+refman-pdf:
 	dune build @refman-pdf
 
-stdlib-html: voboot
+stdlib-html:
 	dune build @stdlib-html
 
-apidoc: voboot
+apidoc:
 	dune build $(DUNEOPT) @doc
 
-release: voboot
+release:
+	@echo "release target is deprecated, use dune directly"
 	dune build $(DUNEOPT) -p coq
 
-fmt: voboot
-	dune build @fmt --auto-promote
+# We define this target as to override Make's built-in one
+install:
+	@echo "To install Coq using dune, use 'dune install -p PACKAGE' where"
+	@echo "PACKAGE is any of the packages defined by opam files in the root dira"
+
+fmt:
+	dune build @fmt
```

Comment:
You recently added the `--auto-promote`. Any reason to backtrack now?

Reply:
Good catch; merge error.